snuba/datasets/schemas/join_schema.py

```diff
@@ -0,0 +1,90 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Mapping, Optional, Sequence
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.datasets.schemas import Schema
+
+
+class JoinType(Enum):
+    INNER = "INNER"
+    LEFT = "LEFT"
+    RIGHT = "RIGHT"
+    FULL = "FULL"
+
+
+@dataclass(frozen=True)
+class JoinMapping():
+    left_alias: str
+    left_column: str
```

Comment:
Would it make sense to put `alias` and `column` into a `NamedTuple` and just have `left` and `right` here that have `left.column`, etc. instead of underscore prefixes?